Error: invalid in patch location

I'm getting below error while creating image tool, could you please check why the error is occurring. I am tryin to build latest wls image with no internet option

```
cache.dir=/jenkins/wls_img_cache
32097173_12.2.1.3.0=/jenkins/stage/p32097173_12213201001_Generic.zip
jdk_8u271=/jenkins/server-jre-8u271-linux-x64.tar.gz
wls_12.2.1.3.0=/jenkins/stage/fmw_12.2.1.3.0_wls.jar
27774698_12.2.1.3.0=/jenkins/stage/p27774698_122130_Generic.zip
28186730_13.9.4.2.4=/jenkins/stage/p28186730_139424_Generic.zip
29637821_12.2.1.3.0=/jenkins/stage/p29637821_122130_Generic.zip
31961038_12.2.1.3.0=/jenkins/stage/p31961038_122130_Generic.zip
```
Error logs

```
[Nov 16, 2020 10:21:07 PM] [INFO]   Deleted the directory "/u01/oracle/.patch_storage/patch_unzip/31961038/files"
[Nov 16, 2020 10:21:07 PM] [INFO]   Deleted the file "/u01/oracle/.patch_storage/patch_unzip/31961038/README.html"
[Nov 16, 2020 10:21:07 PM] [INFO]   Deleted the directory "/u01/oracle/.patch_storage/patch_unzip/31961038"
[Nov 16, 2020 10:21:07 PM] [INFO]   Deleted the directory "/u01/oracle/.patch_storage/patch_unzip"
[Nov 16, 2020 10:21:07 PM] [INFO]   Finishing UtilSession at Mon Nov 16 22:21:07 GMT 2020
[Nov 16, 2020 10:21:07 PM] [INFO]   Log file location: /u01/oracle/cfgtoollogs/opatch/opatch2020-11-16_22-20-49PM_1.log
[Nov 16, 2020 10:21:07 PM] [INFO]   Stack Description: java.lang.RuntimeException: The patch type is invalid in patch location: /u01/oracle/.patch_storage/patch_unzip/6880880/
                                    	at oracle.glcm.opatch.common.impl.PatchFactoryImpl.createPatch(PatchFactoryImpl.java:42)
                                    	at oracle.opatch.opatchutil.CommonAPIUtil.getCommonAPIPatch(CommonAPIUtil.java:80)
                                    	at oracle.opatch.opatchutil.OPatchUtilHelper.validatePatchSubType(OPatchUtilHelper.java:2042)
                                    	at oracle.opatch.opatchutil.OPatchUtilHelper.getPatchesToApply(OPatchUtilHelper.java:911)
                                    	at oracle.opatch.opatchutil.OPatchUtilHelper.getPatchesToApply(OPatchUtilHelper.java:1331)
                                    	at oracle.opatch.opatchutil.NApply.legacy_process(NApply.java:534)
                                    	at oracle.opatch.opatchutil.NApply.legacy_process(NApply.java:368)
                                    	at oracle.opatch.opatchutil.NApply.process(NApply.java:348)
                                    	at oracle.opatch.opatchutil.OUSession.napply(OUSession.java:1105)
                                    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                                    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                                    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                                    	at java.lang.reflect.Method.invoke(Method.java:498)
                                    	at oracle.opatch.UtilSession.process(UtilSession.java:355)
                                    	at oracle.opatch.OPatchSession.process(OPatchSession.java:2650)
                                    	at oracle.opatch.OPatch.process(OPatch.java:779)
                                    	at oracle.opatch.OPatch.main(OPatch.java:829)
                                    Caused by: java.lang.RuntimeException: The patch type is invalid in patch location: /u01/oracle/.patch_storage/patch_unzip/6880880/
                                    	... 17 more
                                    Caused by: oracle.glcm.opatch.common.api.PatchNotFoundException: The patch type is invalid in patch location: /u01/oracle/.patch_storage/patch_unzip/6880880/
                                    	... 17 more
Removing intermediate container 398f182a4e93
[SEVERE ] docker command failed with error: The command '/bin/sh -c /u01/oracle/OPatch/opatch napply -silent -oh /u01/oracle -phBaseDir /tmp/imagetool/patches  && test $? -eq 0  && /u01/oracle/OPatch/opatch util cleanup -silent -oh /u01/oracle  || (cat /u01/oracle/cfgtoollogs/opatch/opatch*.log && exit 1)' returned a non-zero code: 1

```

--- a/imagetool/src/main/java/com/oracle/weblogic/imagetool/cachestore/OPatchFile.java
+++ b/imagetool/src/main/java/com/oracle/weblogic/imagetool/cachestore/OPatchFile.java
@@ -76,6 +76,19 @@ public class OPatchFile extends PatchFile {
         return latestVersion;
     }
 
+    /**
+     * Return true if the patchId matches any known OPatch bug numbers.
+     *
+     * @param patchId the patch ID to test
+     * @return true if and only if the patchId matches an OPatch bug number.
+     */
+    public static boolean isOPatchPatch(String patchId) {
+        if (DEFAULT_BUG_NUM.equals(patchId)) {
+            return true;
+        }
+        return patchId != null && patchId.startsWith(DEFAULT_BUG_NUM + CacheStore.CACHE_KEY_SEPARATOR);
+    }
+
     @Override
     public String resolve(CacheStore cacheStore) throws IOException {
         try {

--- a/imagetool/src/main/java/com/oracle/weblogic/imagetool/cli/menu/CommonOptions.java
+++ b/imagetool/src/main/java/com/oracle/weblogic/imagetool/cli/menu/CommonOptions.java
@@ -295,7 +295,7 @@ public abstract class CommonOptions {
         // add user-provided patch list to any patches that were found for latestPsu or recommendedPatches
         for (String patchId : patches) {
             // if user mistakenly added the OPatch patch to the WLS patch list, skip it
-            if (OPatchFile.DEFAULT_BUG_NUM.equals(patchId)) {
+            if (OPatchFile.isOPatchPatch(patchId)) {
                 continue;
             }
             // if patch ID was provided as bugnumber_version, split the bugnumber and version strings

--- a/imagetool/src/test/java/com/oracle/weblogic/imagetool/cachestore/OPatchFileTest.java
+++ b/imagetool/src/test/java/com/oracle/weblogic/imagetool/cachestore/OPatchFileTest.java
@@ -6,6 +6,7 @@ package com.oracle.weblogic.imagetool.cachestore;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @Tag("unit")
@@ -14,5 +15,9 @@ public class OPatchFileTest {
     void opatchPatchIdTest() {
         assertTrue(OPatchFile.isOPatchPatch("28186730"), "OPatch bug number");
         assertTrue(OPatchFile.isOPatchPatch("28186730_13.9.4.2.4"), "OPatch bug number with version");
+        assertTrue(OPatchFile.isOPatchPatch("28186730_1"), "OPatch bug number with separator");
+
+        assertFalse(OPatchFile.isOPatchPatch("28186731"), "Should not match");
+        assertFalse(OPatchFile.isOPatchPatch("281867301"), "OPatch bug number with additional number");
     }
 }

--- a/imagetool/src/test/java/com/oracle/weblogic/imagetool/cachestore/OPatchFileTest.java
+++ b/imagetool/src/test/java/com/oracle/weblogic/imagetool/cachestore/OPatchFileTest.java
@@ -1,0 +1,18 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package com.oracle.weblogic.imagetool.cachestore;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Tag("unit")
+public class OPatchFileTest {
+    @Test
+    void opatchPatchIdTest() {
+        assertTrue(OPatchFile.isOPatchPatch("28186730"), "OPatch bug number");
+        assertTrue(OPatchFile.isOPatchPatch("28186730_13.9.4.2.4"), "OPatch bug number with version");
+    }
+}

